enhancements/microshift/microshift-router-configuration.md

```diff
@@ -0,0 +1,480 @@
+  policyTypes:
+  - Ingress
+```
+Assuming the pod network is configured to use CIDR `10.42.0.0/16`, the above
+policy will block all traffic not coming from that network.
+
+Any of these methods is equivalent to using firewalld to block traffic, but
+using the CNI instead of a host level setting. This provides greater
+flexibility, as it is self contained within MicroShift APIs.
+
+If a user needs to block traffic to a port this needs a `NetworkPolicy`
+resource like the one shown above. This is NOT automatically done by
+MicroShift, user action is required.
+
+#### Exposing the router
+Relying on [LoadBalancer controller enhancement](loadbalancer-service-support.md)
+means the services are updated by this component. However, ovnk only turns IP
+addresses into iptables rules, it does not take Hostname, which is another
+valid field for the service status.
+
+Since the configuration uses NIC names and also host names, the controller
+needs to translate these to their corresponding IP addresses, and then
+configure them in the service. To keep up with the dynamic nature of IP
+addresses and names, it needs to perform periodic checks and updates.
+
+#### Allowing specific IP addresses
+Allowing specific IP addresses is achieved through the use of `NetworkPolicy`
+resources, as seen in previous examples. These policies include the allowed IP
+addresses and ports in the spec, meaning they reject everything else that is
+not included explicitly.
+
+Due to MicroShift's networking setup, there is one more change required for
+the `NetworkPolicy` to filter out based on the source IP.
+
+As it can be seen in the iptables rules, traffic towards the router ports is
+subject to DNAT to the router service IP and port. All services use
+`externalTrafficPolicy: Cluster`, which will SNAT traffic's source IP to match
+that of the node. This gets further NAT, as described here. By the time the
+`NetworkPolicy` is evaluated the source IP has been replaced with
+`100.64.0.0/16`, which will not produce matches against the configured IP in
+the policy, and traffic will get rejected.
+
+To overcome this, all LoadBalancer services in MicroShift that are subject to
+get NetworkPolicy resources to control external traffic must set
+`externalTrafficPolicy: Local`. This setting will not SNAT packets and the
+source IP is preserved until NetworkPolicy checks run.
+
+This setting, however, comes at the expense of an even load when having more
+than one node, as traffic will not be sent to another node on the same port.
+This is not an issue in single node, which is MicroShift's use case and design
+principle.
+
+This setting is also the default in OCP when deploying in cloud, having a load
+balancer before the cluster. Manifests for both OCP and MicroShift are
+compatible in this regard.
+
+## Open Questions
+N/A
+
+## Test Plan
+All of the changes listed here will be included in the current e2e scenario
+testing harness in MicroShift.
```

Comment:
We should add a test to ensure we're solving the issue described in https://redhat-internal.slack.com/archives/C03DP9PABNC/p1708586503890529 where a service on the cluster is overriding the iptable rule for the router.